src/main/java/org/victorrobotics/frc/dtlib/actuator/motor/ctre/DTTalonFX.java

```diff
@@ -0,0 +1,297 @@
+
+    /* Read config params */
+
+    @Override
+    public boolean isOutputInverted() {
+        return internal.getInverted();
+    }
+
+    @Override
+    public boolean isBrakeEnabled() {
+        return brakeEnabled;
+    }
+
+    @Override
+    public double getPIDproportional() {
+        return pidP;
+    }
+
+    @Override
+    public double getPIDintegral() {
+        return pidI;
+    }
+
+    @Override
+    public double getPIDderivative() {
+        return pidD;
+    }
+
+    @Override
+    public double getPIDfeedforward() {
+        return pidF;
+    }
+
+    @Override
+    public double getPIDintegralZone() {
+        return pidIZ;
+    }
+
+    /* Set motor status */
+
+    @Override
+    public void setPercentOutput(double percent) {
+        internal.set(TalonFXControlMode.PercentOutput, percent);
+    }
+
+    @Override
+    public void setPosition(double position) {
+        internal.set(TalonFXControlMode.Position, position * TICKS_PER_REV);
+    }
+
+    @Override
+    public void setVelocity(double velocity) {
+        internal.set(TalonFXControlMode.Velocity,
+                velocity * TICKS_PER_REV / SECONDS_PER_MINUTE * 0.1);
+    }
+
+    @Override
+    public void neutralOutput() {
+        internal.neutralOutput();
+    }
+
+    @Override
+    public void setEncoderPosition(double position) {
+        internal.setSelectedSensorPosition(position * TICKS_PER_REV);
+    }
+
+    /* Read motor status */
+
+    @Override
+    public double getMotorOutputPercent() {
+        return internal.getMotorOutputPercent();
+    }
+
+    public double getCurrentDraw() {
+        return internal.getSupplyCurrent();
+    }
+
+    @Override
+    public double getTemperature() {
+        return internal.getTemperature();
+    }
+
+    @Override
+    public double getEncoderPosition() {
+        return internal.getSelectedSensorPosition() / TICKS_PER_REV;
+    }
+
+    @Override
+    public double getEncoderVelocity() {
+        return internal.getSelectedSensorVelocity() / TICKS_PER_REV * 10;
+    }
+
+    public DTTalonFXFaults getFaults() {
+        Faults faults = new Faults();
+        internal.getFaults(faults);
+        return new DTTalonFXFaults(faults);
+    }
+
+    @Override
+    public String getFirmwareVersion() {
+        if (firmwareVersion == null) {
+            int version = internal.getFirmwareVersion();
+            firmwareVersion = (version >> 8) + "." + (version & 0xFF);
+        }
+        return firmwareVersion;
+    }
+
+    @Override
+    public void close() {
+        internal.close();
+    }
+
+    @Override
+    public void customizeSendable(SendableBuilder builder) {
+        builder.addDoubleProperty("Current", this::getCurrentDraw, null);
+    }
+
+    @Override
+    public int getCanID() {
+        return deviceID;
+    }
+
+    @Override
+    public double getInputVoltage() {
+        return internal.getBusVoltage();
+    }
+
+    public static class DTTalonFXFaults implements DTMotorFaults<Faults> {
```

Comment:
If we can avoid parameterizing the interface (as with the motor), that would be helpful. That might just involve moving the `internal` method into the concrete class (this one).

Reply:
Again, clients don't need to parameterize their objects if they don't need the `internal()` functionality. I'm not sure how the parameter would be removed here.